web/app/js/components/Tap.jsx

```diff
@@ -92,47 +166,264 @@ class Tap extends React.Component {
+          </Col>
+        </Row>
+
+        <Row gutter={rowGutter}>
+          <Col span={2 * colSpan}>
+            <Form.Item
+              extra="Display requests with this :authority">
+              <Input placeholder="Authority" onChange={this.handleFormEvent("authority")} />
+            </Form.Item>
+          </Col>
+
+          <Col span={2 * colSpan}>
+            <Form.Item
+              extra="Display requests with paths that start with this prefix">
+              <Input placeholder="Path" onChange={this.handleFormEvent("path")} />
+            </Form.Item>
+          </Col>
+
+        </Row>
+
+        <Row gutter={rowGutter}>
+          <Col span={colSpan}>
+            <Form.Item
+              extra="Display requests with this scheme">
+              <Input placeholder="Scheme" onChange={this.handleFormEvent("scheme")} />
+            </Form.Item>
+          </Col>
+
+          <Col span={colSpan}>
+            <Form.Item
+              extra="Maximum requests per second to tap">
+              <Input
+                defaultValue={defaultMaxRps}
+                placeholder="Max RPS"
+                onChange={this.handleFormEvent("maxRps")} />
+            </Form.Item>
+          </Col>
+
+          <Col span={colSpan}>
+            <Form.Item
+              extra="Display requests with this HTTP method">
+              <Select
+                allowClear
+                placeholder="HTTP method"
+                onChange={this.handleFormChange("method")}>
+                {
+                  _.map(httpMethods, m =>
+                    <Select.Option key={`method-select-${m}`} value={m}>{m}</Select.Option>
+                  )
+                }
+              </Select>
+            </Form.Item>
+          </Col>
+        </Row>
+      </React.Fragment>
+    );
+  }
+
+  renderCurrentQuery = () => {
+    let emptyVals =  _.countBy(_.values(this.state.query), v => _.isEmpty(v));
+
+    return (
+      <div className="tap-query">
+        <code>
+          { !emptyVals.false || emptyVals.false === 1 ? null :
+          <div>
+            Current query:
+            {
+              _.map(this.state.query, (query, queryName) => {
+                if (query === "") {
+                  return null;
+                }
+                return <div key={queryName}>{queryName}: {query}</div>;
+              })
+            }
+          </div>
+          }
+        </code>
+      </div>
+    );
+  }
+
   render() {
     return (
       <div>
-        {!this.state.errors ? null : <ErrorBanner message={this.state.errors} />}
+        {!this.state.error ? null :
+        <ErrorBanner message={this.state.error} onHideMessage={() => this.setState({ error: "" })} />}
 
         <PageHeader header="Tap" />
         {this.renderTapForm()}
+        {this.renderCurrentQuery()}
 
         <div className="tap-display">
           <code>
             { this.state.webSocketRequestSent && _.size(this.state.messages) === 0 ?
```

Comment:
TIOLI With the additional filters, it's easier to construct a tap query that doesn't return any results, in which case we'll display the loading spinner indefinitely. I think it might be better to distinguish between the state where the request is still pending, and the state where a response has been received but it doesn't return any results, and in the latter case display a "no results" message instead of a spinner.

Reply:
yeah! good point!